src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsVehicleRentalDataSource.java

```diff
@@ -31,214 +34,144 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(GbfsVehicleRentalDataSource.class);
 
-    private static final String DEFAULT_NETWORK_NAME = "GBFS";
+    private final String url;
 
-    // station_information.json required by GBFS spec
-    private final GbfsStationDataSource stationInformationSource;
+    private final String language;
 
-    // station_status.json required by GBFS spec
-    private final GbfsStationStatusDataSource stationStatusSource;
-
-    // free_bike_status.json declared OPTIONAL by GBFS spec
-    private final GbfsFloatingVehicleDataSource floatingBikeSource;
-
-    private final String networkName;
+    private final Map<String, String> httpHeaders;
 
     /** Some car rental systems and flex transit systems work exactly like bike rental, but with cars. */
     private final boolean routeAsCar;
 
+    private final boolean allowKeepingRentedVehicleAtDestination;
+
+    private GbfsFeedLoader loader;
+
     public GbfsVehicleRentalDataSource(GbfsVehicleRentalDataSourceParameters parameters) {
+        url = parameters.getUrl();
+        language = parameters.language();
+        httpHeaders = parameters.getHttpHeaders();
         routeAsCar = parameters.routeAsCar();
-        stationInformationSource = new GbfsStationDataSource(parameters);
-        stationStatusSource = new GbfsStationStatusDataSource(parameters);
-        floatingBikeSource = OTPFeature.FloatingBike.isOn()
-            ? new GbfsFloatingVehicleDataSource(parameters)
-            : null;
-
-        configureUrls(parameters.getUrl(), parameters.getHttpHeaders());
-        this.networkName = parameters.getNetwork(DEFAULT_NETWORK_NAME);
+        allowKeepingRentedVehicleAtDestination = parameters.allowKeepingRentedVehicleAtDestination();
     }
 
-    private void configureUrls(String url, Map<String, String> headers) {
-        GbfsAutoDiscoveryDataSource gbfsAutoDiscoveryDataSource = new GbfsAutoDiscoveryDataSource(url, headers);
-        stationInformationSource.setUrl(gbfsAutoDiscoveryDataSource.stationInformationUrl);
-        stationStatusSource.setUrl(gbfsAutoDiscoveryDataSource.stationStatusUrl);
-        if (OTPFeature.FloatingBike.isOn()) {
-            floatingBikeSource.setUrl(gbfsAutoDiscoveryDataSource.freeBikeStatusUrl);
-        }
+    @Override
+    public void setup() {
+        loader = new GbfsFeedLoader(url, httpHeaders, language);
```

Comment:
I really like that you are not doing this in the constructor. It is a best-practice to avoid "risky heavy lifting" activities in the constructor.